tests/scanner/audit/retention_rules_engine_test.py

```diff
@@ -851,6 +851,130 @@ def test_bucket_on_multi_project_has_vio(self):
             fake_bucket, 'bucket retention on multi projects')
         self.assertEqual(got_violations, expected_violations)
 
+    yaml_str_bigquery_retention_on_projects = """
+rules:
+  - name: bigquery retention on projects
+    applies_to:
+      - bigquery_table
+    resource:
+      - type: project
+        resource_ids:
+          - def-project-5
+    minimum_retention: 90
+    maximum_retention: 90
+
+"""
+    def test_bigquery_retention_on_project_no_expiration_time(self):
+        """Test a bigquery table without expiration time."""
+        rules_engine = get_rules_engine_with_rule(RetentionRulesEngineTest.yaml_str_bigquery_retention_on_projects)
+        self.assertTrue(1 <= len(rules_engine.rule_book.resource_rules_map))
+
+        data_creater = frsd.FakeTableDataCreater('fake_bqtable', frsd.DATASET1)
+
+        fake_table = data_creater.get_resource()
+        got_violations = list(rules_engine.find_violations(fake_table))
+        expected_violations = frsd.build_table_violations(
+            fake_table, 'bigquery retention on projects')
+        self.assertEqual(got_violations, expected_violations)
+
+    def test_bigquery_retention_on_project_too_big(self):
+        """Test that a rule with a resource.type equal to 'project'"""
```

Comment:
Update doc strings (here and below) to cover what "too big", "too small", etc mean.